src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,160 @@
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.simulation.BatterySim;
+import edu.wpi.first.wpilibj.simulation.EncoderSim;
+import edu.wpi.first.wpilibj.simulation.RoboRioSim;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import frc.robot.commands.ArmAutoCommand;
+
+
+public class ArmSubsystem extends SubsystemBase implements AutoCloseable {
+
+  // FIXME using PWMSparkMax because CANSparkMax doesn't have an equivalent simulation class
+  // FIXME https://docs.wpilib.org/en/stable/docs/software/hardware-apis/sensors/encoders-software.html
+  // May limit how much we can do in terms of JUnit tests
+  //Base arm PID values
+  private double BP = 1;
+  private double BI = 0;
+  private double BD = 0.1;
+  //Elbow arm PID values
+  private double EP = 0.05;
+  private double EI = 0;
+  private double ED = 0.2;
+  private CANSparkMax motorBaseOne;
+  private CANSparkMax motorBaseTwo;
+  private CANSparkMax motorElbow;
+  private final Encoder motorBaseOneEncoder;
+  private final EncoderSim motorBaseOneEncoderSim;
+  private final Encoder motorElbowEncoder;
+  private final EncoderSim motorElbowEncoderSim;
+  private final PIDController basePIDController;
+  private final PIDController elbowPIDController;
+  private ShuffleboardTab configTab = Shuffleboard.getTab("Arm");
+  //Base Arm Values for Simulation
+  private GenericEntry simBArmAngle = configTab.add("SimBase Arm Angle", 0.0).getEntry();
+  private GenericEntry simBEncoderPos = configTab.add("SimBase Encoder Angle", 0.0).getEntry();
+  private GenericEntry simBOutSet = configTab.add("SimBase Output", 0.0).getEntry();
+  private GenericEntry simBError = configTab.add("SimBase Error", 0.0).getEntry();
+  private GenericEntry simBVoltage = configTab.add("SimBase Motor Voltage", 0.0).getEntry();
+  //Elbow Arm Values for Simulation
+  private GenericEntry simEArmAngle = configTab.add("SimElbow Arm Angle", 0.0).getEntry();
+  private GenericEntry simEEncoderPos = configTab.add("SimElbow Encoder Angle", 0.0).getEntry();
+  private GenericEntry simEOutSet = configTab.add("SimElbow Output", 0.0).getEntry();
+  private GenericEntry simEError = configTab.add("SimElbow Error", 0.0).getEntry();
+  private GenericEntry simEVoltage = configTab.add("SimElbow Motor Voltage", 0.0).getEntry();
+  
+
+  // SIM VALUES:
+  private double baseGearing = 15.;
+  private double elbowGearing = 8.57142857;
+  public static double baseArmLength = Units.inchesToMeters(43.5);
+  public static double elbowArmLength = Units.inchesToMeters(37.5);
+  private double minBAngle = Units.degreesToRadians(45);
+  private double maxBAngle = Units.degreesToRadians(91);
+  private double minEAngle = Units.degreesToRadians(-70);
+  private double maxEAngle = Units.degreesToRadians(89);
+  private double baseArmMass = Units.lbsToKilograms(15); //15
+  private double elbowArmMass = Units.lbsToKilograms(15);
```

Comment:
all of these constants should be declared as public static final, should be moved to the first lines of the subsystem, and should start with a k prefix

Reply:
Done